pkg/apis/core/validation/validation.go

```diff
@@ -74,6 +74,8 @@ var iscsiInitiatorIqnRegex = regexp.MustCompile(`iqn\.\d{4}-\d{2}\.([[:alnum:]-.
 var iscsiInitiatorEuiRegex = regexp.MustCompile(`^eui.[[:alnum:]]{16}$`)
 var iscsiInitiatorNaaRegex = regexp.MustCompile(`^naa.[[:alnum:]]{32}$`)
 
+var csiDriverNameRexp = regexp.MustCompile(`^[A-Za-z]+(\.?-?_?[A-Za-z0-9-])+$`)
```

Comment:
I'm unsure this regex matches https://github.com/container-storage-interface/spec/blob/master/spec.md#getplugininfo

This would allow `foo.-_bar`... is that intentional?

Reply:
I think it doesn't match https://github.com/container-storage-interface/spec/blob/master/spec.md#getplugininfo.
It is origin from `csi_plugin.go`, and I moved it to here.
@vladimirvivien can you take a look?